src/bodywork/workflow_execution.py

```diff
@@ -42,53 +42,34 @@
     BodyworkNamespaceError,
     BodyworkDockerImageError,
     BodyworkGitError,
+    BodyworkConfigError,
 )
 from .git import download_project_code_from_repo, get_git_commit_hash
 from .logs import bodywork_log_factory
 
 _log = bodywork_log_factory()
 
 
-def get_config_from_git_repo(
-    repo_url: str,
-    repo_branch: str = "master",
-    cloned_repo_dir: Path = DEFAULT_PROJECT_DIR,
-) -> BodyworkConfig:
-    """Download project's Git repo and parse config.
-
-    :param repo_url: Git repository URL.
-    :param repo_branch: The Git branch to download, defaults to 'master'.
-    :param cloned_repo_dir: The name of the directory into which the
-        repository will be cloned, defaults to DEFAULT_PROJECT_DIR.
-    """
-    try:
-        download_project_code_from_repo(repo_url, repo_branch, cloned_repo_dir)
-        path_to_project_config_file = cloned_repo_dir / PROJECT_CONFIG_FILENAME
-        config = BodyworkConfig(
-            path_to_project_config_file, check_py_modules_exist=True
-        )
-    finally:
-        rmtree(cloned_repo_dir, onerror=_remove_readonly)
-    return config
-
-
 def run_workflow(
-    config: BodyworkConfig,
     namespace: str,
     repo_url: str,
     repo_branch: str = "master",
     docker_image_override: Optional[str] = None,
+    config: Optional[BodyworkConfig] = None,
```

Comment:
if you rename this `config_override` then,

```python
        config = (
            config_override
            if config_override is not None
            else BodyworkConfig(cloned_repo_dir / PROJECT_CONFIG_FILENAME, True)
        )
```

Should solve all you problems (have checked it at my end).

Reply:
I you don't like the name `config_override` then you could use `config` in the argument and `config_` within the body of the function?

Reply:
I you don't like the name `config_override` then you could use `config` in the argument and `config_` within the body of the function?

Reply:
If this fixes the false positives from MyPy then i'm for it. That's poor form for Python that the type checker doesn't work properly but I suppose that's what you get with a language that isn't strongly typed.

Reply:
MyPy is still relatively new and is constantly being updated. Guido von Rossum looks after the project personally.